src/libraries/System.Net.Http/src/System/Net/Http/HttpTelemetry.cs

```diff
@@ -0,0 +1,104 @@
+
+        public static new bool IsEnabled => Log.IsEnabled();
+
+        // NOTE
+        // - The 'Start' and 'Stop' suffixes on the following event names have special meaning in EventSource. They
+        //   enable creating 'activities'.
+        //   For more information, take a look at the following blog post:
+        //   https://blogs.msdn.microsoft.com/vancem/2015/09/14/exploring-eventsource-activity-correlation-and-causation-features/
+        // - A stop event's event id must be next one after its start event.
+
+        [Event(1, Level = EventLevel.Informational)]
+        public void RequestStart(string host, int port)
+        {
+            Interlocked.Increment(ref _startedRequests);
+            Interlocked.Increment(ref _currentRequests);
+            WriteEvent(1, host, port);
+        }
+
+        [Event(2, Level = EventLevel.Informational)]
+        public void RequestStop(string host, int port)
+        {
+            Interlocked.Decrement(ref _currentRequests);
+            WriteEvent(2, host, port);
+        }
+
+        [NonEvent]
+        public void RequestAbort()
+        {
+            Interlocked.Increment(ref _abortedRequests);
+        }
+
+        protected override void OnEventCommand(EventCommandEventArgs command)
+        {
+            if (command.Command == EventCommand.Enable)
+            {
+                // This is the convention for initializing counters in the RuntimeEventSource (lazily on the first enable command).
+                // They aren't disabled afterwards...
+
+                // The cumulative number of HTTP requests started since the process started.
+                _startedRequestsCounter ??= new PollingCounter("requests-started", this, () => _startedRequests)
+                {
+                    DisplayName = "Requests Started",
+                };
+
+                // The number of HTTP requests started per second since the process started.
+                _requestsPerSecondCounter ??= new IncrementingPollingCounter("requests-started-per-second", this, () => _startedRequests)
+                {
+                    DisplayName = "Requests Started Rate",
+                    DisplayRateTimeScale = TimeSpan.FromSeconds(1)
+                };
+
+                // The cumulative number of HTTP requests aborted since the process started.
+                // Aborted means that an exception occurred during the handler's Send(Async) call as a result of a
+                // connection related error, timeout, or explicitly cancelled.
+                _abortedRequestsCounter ??= new PollingCounter("requests-aborted", this, () => _abortedRequests)
+                {
+                    DisplayName = "Requests Aborted"
+                };
+
+                // The number of HTTP requests aborted per second since the process started.
+                _requestsPerSecondCounter ??= new IncrementingPollingCounter("requests-aborted-per-second", this, () => _abortedRequests)
```

Comment:
These per-second counters can be derived from their absolute variants. Do these take extra CPU to log?

Reply:
@noahfalk Can you also comment on this? I think the overhead of this counter is small.

Reply:
The performance overhead of a polling counter should be extremely low and it scales based on how frequently the telemetry listener wants to poll it. What will happen is every N seconds (the telemetry listener chooses N) your callback function will be invoked, whatever value it returns will be serialized into a message and that message will be emitted into the event stream. Assuming your callback is not CPU intensive this should translate into a few allocations and maybe maybe 1us of CPU activity every N seconds. 
Likely the biggest performance cost to be aware of is the cost the library pays to update the _abortedRequests field on each request, but you are fully in control of that overhead.

Reply:
Thanks @noahfalk for the feedback. Based on this I'll leave these separate counters.